account_reconciliation_widget/models/reconciliation_widget.py

```diff
@@ -115,10 +115,10 @@ def get_move_lines_for_bank_statement_line(
             """
             SELECT "account_move_line".id, COUNT(*) OVER() FROM {from_clause}
             {where_str}
-            ORDER BY ("account_move_line".debit -
-                      "account_move_line".credit) = {amount} DESC,
-                "account_move_line".date_maturity ASC,
-                "account_move_line".id ASC
+            ORDER BY "account_move_line".date_maturity DESC,
+                    ("account_move_line".debit -
+                    "account_move_line".credit) = {amount} DESC,
```

Comment:
Don't put down this. It's very useful to have first the exact amount matches

Reply:
I moved {amount} to the second line to have the correct behavior, if it remains in the first GROUP BY, the newest due dates are brought but even so there are still old dates present, so moving it to second position and changing the others to DESC, we have behavior similar to v12

Reply:
How could we deal with this conflict without moving it to second position?

Reply:
As said, I prefer to have first the exact matches, no matter the maturity date.

Reply:
I don't know how to resolve it, but as is right now it's very useful and I don't want to lose this feature (in fact, I miss it a lot in v16).

Reply:
I understand, at this point I will perform the tests again and upload the commit again, perhaps it could have been some cache in my environment that caused this impression of conflict and made me move to second position

Reply:
After making the adjustments, it had the same behavior, possibly some momentary cache made me have this impression and the need to change it to second position